sound/soc/sof/core.c

```diff
@@ -379,6 +379,10 @@ static int sof_probe(struct platform_device *pdev)
 static int sof_remove(struct platform_device *pdev)
 {
 	struct snd_sof_dev *sdev = dev_get_drvdata(&pdev->dev);
+	struct snd_sof_pdata *pdata = sdev->pdata;
+
+	if (pdata && !IS_ERR(pdata->pdev_mach))
+		platform_device_unregister(pdata->pdev_mach);
```

Comment:
@plbossart for my education: we also unregister the machine driver in sof_pci_remove(). Is it required to have it in both places?

Reply:
@ranj063 very nice catch. This makes absolutely no sense and the initial code doesn't seem to do what it's supposed to do. Thanks for reporting this, very useful.

Reply:
I tracked all the problems in https://github.com/thesofproject/linux/issues/259. new PR pushed with all duplicate code removed.